### PR TITLE
mypy: Exclude only 20 files that are still failing

### PR DESCRIPTION
**Pull request by cclauss**
Let's see what is required to finish  #4052

### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by spazm** [comment]
`other/` will be clean when the PRs in-fligh are landedt: #5647, #5648, #5653, #5656, and  #5755.

The middle three were autoclosed by the bot for "multiple PR detected", so if they could be re-opened and reviewed, that'd be appreciated.  [ #5648 #5653 #5656 ]

```
% git checkout mypy-fix-other-all

% mypy --ignore-missing-imports --install-types
Success: no issues found in 21 source files
```

**Comment by dylanbuchi** [comment]
> The middle three were autoclosed by the bot for "multiple PR detected", so if they could be re-opened and reviewed, that'd be appreciated. [ #5648 #5653 #5656 ]

Hey @spazm. PR's can't be reopened! You'll have to create new pull requests.

**Comment by spazm** [comment]
> Hey @spazm. PR's can't be reopened! You'll have to create new pull requests.

Thanks for the info.  It's a bummer, of course.

**Comment by cclauss** [comment]
Improvements!  
```
./graphs          || true  # Errors: 10  -> 9
./maths           || true  # Errors: 14  -> 14
./matrix          || true  # Errors: 15  -> 15
./searches        || true  # Errors: 45  -> 21
./other           || true  # Errors: 51  -> 43
./data_structures || true  # Errors: 117 -> 67
```
Perhaps we should start `mypy --exclude` individual files.

**Comment by cclauss** [comment]
```python
with open("mypy_errors.txt") as out_file:
    files = sorted({line.split(':')[0] for line in in_file if ':' in line})

print(f"{len(files) = }")
print("\n".join(f"                  --exclude={name}" for name in names))
```

**Comment by cclauss** [comment]
@dhruvmanila Please let me know why the `mypy.ini` __exclude__ is not working.

**Comment by ErwinJunge** [comment]
> @dhruvmanila Please let me know why the `mypy.ini` **exclude** is not working.

Probably due to the `mv mypy.ini was_mypy.ini` here: https://github.com/TheAlgorithms/Python/pull/5608/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R31

**Comment by cclauss** [comment]
I was talking about the mypy run on line 24.

File: mypy.ini [hunk 1]
```diff
@@ -2,4 +2,4 @@
...
-exclude = (data_structures/stacks/next_greater_element\.py|graphs/boruvka\.py|graphs/breadth_first_search\.py|graphs/breadth_first_search_2\.py|graphs/check_cycle\.py|graphs/finding_bridges\.py|graphs/greedy_min_vertex_cover\.py|graphs/random_graph_generator\.py|maths/average_mode\.py|maths/gamma_recursive\.py|maths/proth_number\.py|maths/series/geometric_series\.py|maths/series/p_series\.py|matrix/matrix_operation\.py|other/fischer_yates_shuffle\.py|other/least_recently_used\.py|other/lfu_cache\.py|other/lru_cache\.py|searches/simulated_annealing\.py|searches/ternary_search\.py)/$
+exclude = /(data_structures/stacks/next_greater_element\.py|graphs/boruvka\.py|graphs/breadth_first_search\.py|graphs/breadth_first_search_2\.py|graphs/check_cycle\.py|graphs/finding_bridges\.py|graphs/greedy_min_vertex_cover\.py|graphs/random_graph_generator\.py|maths/average_mode\.py|maths/gamma_recursive\.py|maths/proth_number\.py|maths/series/geometric_series\.py|maths/series/p_series\.py|matrix/matrix_operation\.py|other/fischer_yates_shuffle\.py|other/least_recently_used\.py|other/lfu_cache\.py|other/lru_cache\.py|searches/simulated_annealing\.py|searches/ternary_search\.py)$
```
Comment (dylanbuchi) [inline mypy.ini:5]: @cclauss 
This should work
```suggestion
exclude = (data_structures/stacks/next_greater_element.py|graphs/boruvka.py|graphs/breadth_first_search.py|graphs/breadth_first_search_2.py|graphs/check_cycle.py|graphs/finding_bridges.py|graphs/greedy_min_vertex_cover.py|graphs/random_graph_generator.py|maths/average_mode.py|maths/gamma_recursive.py|maths/proth_number.py|maths/series/geometric_series.py|maths/series/p_series.py|matrix_operation.py|other/fischer_yates_shuffle.py|other/least_recently_used.py|other/lfu_cache.py|other/lru_cache.py|searches/simulated_annealing.py|searches/ternary_search.py)

```

**Review by dylanbuchi — CHANGES_REQUESTED** [review]

**Comment by cclauss** [comment]
@dylanbuchi AWESOME!!  I own you a beer!!

File: mypy.ini [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [mypy]
 ignore_missing_imports = True
+install_types = True
+non_interactive = True
+exclude = (data_structures/stacks/next_greater_element.py|graphs/boruvka.py|graphs/breadth_first_search.py|graphs/breadth_first_search_2.py|graphs/check_cycle.py|graphs/finding_bridges.py|graphs/greedy_min_vertex_cover.py|graphs/random_graph_generator.py|maths/average_mode.py|maths/gamma_recursive.py|maths/proth_number.py|maths/series/geometric_series.py|maths/series/p_series.py|matrix_operation.py|other/fischer_yates_shuffle.py|other/least_recently_used.py|other/lfu_cache.py|other/lru_cache.py|searches/simulated_annealing.py|searches/ternary_search.py)
```
Comment (dhruvmanila) [inline mypy.ini:5]: Maybe this could be done on multiline if possible for better readability. If not possible, then ignore this comment.

**Review by dhruvmanila — APPROVED** [review]
A small nitpick, otherwise LGTM 👍

**Review by poyea — APPROVED** [review]

**Pull request merged**